src/proxy/mod.rs

```diff
@@ -145,18 +145,31 @@ pub async fn copy_hbone(
     let client_to_server = async {
         let mut ri = tokio::io::BufReader::with_capacity(HBONE_BUFFER_SIZE, &mut ri);
         let mut wo = tokio::io::BufWriter::with_capacity(HBONE_BUFFER_SIZE, &mut wo);
-        let res = tokio::io::copy(&mut ri, &mut wo).await;
-        info!(?res, ?desc, "hbone -> tcp");
-        res.expect("");
-        wo.shutdown().await
+        match tokio::io::copy(&mut ri, &mut wo).await {
+            Ok(len) => {
+                info!(len, ?desc, "hbone -> tcp");
+                wo.shutdown().await
```

Comment:
This is changing, no shutdown when err

Reply:
There was no `shutdown` previously, since it would `panic` on `Err`.

Reply:
OK, I added it back, and just remove the "expect", and use "shutdown" as an error indication.